src/fu-engine.c

```diff
@@ -225,10 +225,32 @@ fu_engine_watch_device (FuEngine *self, FuDevice *device)
 			  G_CALLBACK (fu_engine_status_notify_cb), self);
 }
 
+static void
+fu_engine_ensure_device_battery_inhibit (FuEngine *self, FuDevice *device)
+{
+	if (fu_device_has_flag (device, FWUPD_DEVICE_FLAG_REQUIRE_AC) &&
+	    (fu_context_get_battery_state (self->ctx) == FU_CONTEXT_BATTERY_STATE_DISCHARGING ||
```

Comment:
Not sure I agree with this.  You can have an AC adapter plugged in that doesn't provide enough power so the system is effectively discharging, but really slowly.

Reply:
From a UPower point of view if the AC adapter is connected it's either _charging_ or _charged_. I don't think we handle the _plugged in but discharging_ case at all in the UI.

Reply:
FWIW Windows doesn't really handle this well either.  I think it's most likely fine, but wanted to mention the terms were not strictly true :)

Reply:
Out of morbid curiosity, is there any way to detect this case from the OS without just noticing that the percentage is actually going down over time?